README.md

```diff
@@ -244,39 +244,44 @@ You can also add them to the end of `~/.profile` to set this by default in all s
 
 #### MacOS dependencies installation
 
-To install GCC and CMake on MacOS we recommend to use [Homebrew](https://brew.sh/) package manager. With Homebrew
+First of all, install Xcode Command Line Tools if you don't have them. Open terminal and run:
+```sh
+xcode-select --install
+```
+You'll see a panel that asks you to install Xcode Command Line Tools. Click 'Install' to begin the installation process and wait for it to complete.
+
+To install GCC and CMake on macOS we recommend to use [Homebrew](https://brew.sh/) package manager. With Homebrew
 installed, run the following commands:
 ```sh
-brew install gcc cmake 
+brew install gcc@14 cmake 
 ```
 After installation, check `cmake --version`. If command is not found, then you need to add to environment path to
 homebrew installed packages. To do this open `~/.zprofile` (for Zsh) or
 `~/.bash_profile` (for Bash) and add to the end of the file the output of `brew shellenv`.
 After that, restart the terminal and check the version of CMake again, now it should be displayed.
 
-Then, check the installed version of GCC:`brew info gcc`. You must see something like `==> gcc: stable X.Y.Z ...`. 
-Check that `gcc-X` and `g++-X` commands work, where X is the version from this output 
-(this designation continues to be used further).
-
-After you need to install Boost library. Please, don't use Homebrew for this, as it may not work correctly.
+Then you need to install Boost library built with GCC. Please, don't use Homebrew for this, because brew boost is built with Clang.
 Instead, download the latest version of Boost from the [official website](https://www.boost.org/users/download/).
```

Comment:
And why is it bad that brew boost is built with clang? This needs some explanation
I.e 

> Please avoid using Homebrew for this, as the Boost version provided by Homebrew is built with Clang, which..